merkle/src/proof.rs

```diff
@@ -56,7 +56,7 @@ impl<T: Eq + Clone + AsRef<[u8]>> Proof<T> {
     }
 
     /// Verifies MT inclusion proof and that leaf_data is the original leaf data for which proof was generated.
-    pub fn validate_with_data<A: Algorithm<T>>(&self, leaf_data: &Hashable<A>) -> bool {
+    pub fn validate_with_data<A: Algorithm<T>>(&self, leaf_data: &dyn Hashable<A>) -> bool {
```

Comment:
it sounds to me that that `dyn` will make things slow :S

Reply:
I just made that change to satisfy Clippy. It's only a syntax upgrade:

```
    Checking merkletree v0.7.1 (/Users/clwk/dev/go/src/github.com/filecoin-project/merkle_light/merkle)
warning: trait objects without an explicit `dyn` are deprecated
  --> merkle/src/proof.rs:59:67
   |
59 |     pub fn validate_with_data<A: Algorithm<T>>(&self, leaf_data: &Hashable<A>) -> bool {
   |                                                                   ^^^^^^^^^^^ help: use `dyn`: `dyn Hashable<A>`
   |
   = note: `#[warn(bare_trait_objects)]` on by default

    Finished dev [unoptimized + debuginfo] target(s) in 1.51s
```